quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/integration/QuarkusKeycloakSessionFactory.java

```diff
@@ -99,9 +100,12 @@ public void init() {
         if (componentFactoryPF != null) {
             componentFactoryPF.postInit(this);
         }
+        // Init JpaConnectionProvider first to ensure that DB resources are available to all other providers
+        ProviderFactory<JpaConnectionProvider> jpaUpdaterProvider = getProviderFactory(JpaConnectionProvider.class);
+        jpaUpdaterProvider.postInit(this);
```

Comment:
@keycloak/cloud-native I would like advice here.

With JDBC_PING2, the Infinispan cache manager now has a hard dependency on the JDBCPING DB table and we need to ensure that all DB creations/updates occur before the cache manager is created. The current workflow seems to be that the updates are initiated by `QuarkusJpaUpdaterProvider#update` via `QuarkusJpaConnectionProviderFactory#postInit`, hence my current code which ensures the `postInit` method on this ProviderFactory is called first. This solution works, but seems a bit hacky and I wondered if there was a better approach to solving this issue?

After discussing with @mhajas, I have tried to use Quarkus BuildItem's with the `CacheBuildSteps` to force a before/after relationship with Liquibase and Jpa, but this does not seem to work as ultimately a call to `postInit` on the factory is what initiates the migration.

Any thoughts/suggestions much appreciated :slightly_smiling_face:

Reply:
Think we need some priorities or dependencies between factories, so that we initialize things in the correct order. This is a bit to hacky if you ask me ;)

Reply:
I've added an optional `postInitPriority` method to the `ProviderFactory` which is used to sort the ProviderFactory implementations before executing `postInit`. The upshot is we can also remove the explicit `ComponentFactoryProviderFactory::postInit` call by returning `Integer.MAX_VALUE`. Maybe this is as equally as hacky as my initial attempt though :slightly_smiling_face:

Reply:
Actually that broke a ton of things :sweat_smile: , reverted, I'll wait for input from the Pros.

Reply:
I agree @vmuzikar. I experimented a bit more locally to see why adding a priority to the postInit order did not work and AFAICT it seems that we're relying on the JVM ordering of the HashMap entries atm and it just happens to work, which of course may change in subsequent updates or with different JVMs. So I think a more robust mechanism is definitely required.